src/librustc_passes/ast_validation.rs

```diff
@@ -244,7 +244,12 @@ impl<'a> AstValidator<'a> {
     fn check_trait_fn_not_async(&self, span: Span, asyncness: IsAsync) {
         if asyncness.is_async() {
             struct_span_err!(self.session, span, E0706,
-                             "trait fns cannot be declared `async`").emit()
+                             "trait fns cannot be declared `async`")
+                .note("Due to technical restrictions rust does not currently support `async` \
+                       trait fns.")
```

Comment:
```
                .note("due to technical restrictions Rust does not currently support `async` \
                       trait functions")
```

@nikomatsakis @rust-lang/docs how would we feel about adding a link to https://smallcultfollowing.com/babysteps/blog/2019/10/26/async-fn-in-traits-are-hard/ here? Alternatively, @nikomatsakis, would you feel up to adding that blogpost (or a summarized version of it) to the book or some other stable documentation location?

Reply:
I think we should publish it on the inside Rust blog, perhaps, though I'm not sure if it's quite a "perfect fit." I don't think we should link to random-ish (obviously in this case it's Niko, but the point stands, I think) blogs from inside the compiler. We might also be able to uplift the content into a extended diagnostic perhaps?

Reply:
Extending the error code description sounds like the best course of action.

Reply:
Rather than link to blog posts, we could link to the async await book - https://rust-lang.github.io/async-book/07_workarounds/06_async_in_traits.html
(I'd be more than happy for the async book to be referencing useful blog posts)